jmix-quartz/quartz/src/main/java/io/jmix/quartz/service/QuartzService.java

```diff
@@ -46,66 +47,75 @@ public List<JobModel> getAllJobs() {
             List<JobKey> jobDetailsKeys = jobDetailsFinder.getJobDetailBeanKeys();
 
             for (JobKey jobKey : scheduler.getJobKeys(GroupMatcher.anyJobGroup())) {
+                JobDetail jobDetail;
                 try {
-                    JobDetail jobDetail = scheduler.getJobDetail(jobKey);
-
-                    JobModel jobModel = dataManager.create(JobModel.class);
-                    jobModel.setJobName(jobKey.getName());
-                    jobModel.setJobGroup(jobKey.getGroup());
-                    jobModel.setJobDataParameters(getDataParamsOfJob(jobKey));
-
-                    jobModel.setJobClass(jobDetail.getJobClass().getName());
-                    jobModel.setDescription(jobDetail.getDescription());
-
-                    jobModel.setJobSource(jobDetailsKeys.contains(jobKey) ? JobSource.PREDEFINED : JobSource.USER_DEFINED);
-
-                    List<TriggerModel> triggerModels = new ArrayList<>();
-                    List<? extends Trigger> jobTriggers = scheduler.getTriggersOfJob(jobKey);
-                    if (!CollectionUtils.isEmpty(jobTriggers)) {
-                        boolean isActive = false;
-                        for (Trigger trigger : scheduler.getTriggersOfJob(jobKey)) {
-                            TriggerModel triggerModel = dataManager.create(TriggerModel.class);
-                            triggerModel.setTriggerName(trigger.getKey().getName());
-                            triggerModel.setTriggerGroup(trigger.getKey().getGroup());
-                            triggerModel.setScheduleType(trigger instanceof SimpleTrigger ? ScheduleType.SIMPLE : ScheduleType.CRON_EXPRESSION);
+                    jobDetail = scheduler.getJobDetail(jobKey);
+
+                } catch (JobPersistenceException e) {
+                    if (e.getCause() instanceof ClassNotFoundException) {
+                        jobDetail = new BrokenJobDetail(jobKey);
+                    } else {
+                        continue;
+                    }
+                } catch (SchedulerException e) {
+                    log.error("Unable to fetch information about the job: {}", jobKey, e);
+                    continue;
+                }
+
+                JobModel jobModel = dataManager.create(JobModel.class);
+                jobModel.setJobName(jobKey.getName());
+                jobModel.setJobGroup(jobKey.getGroup());
+                jobModel.setJobDataParameters(getDataParamsOfJob(jobKey));
+
+                jobModel.setJobClass(jobDetail.getJobClass().getName());
```

Comment:
We should display actual missing class. 

- We should be able to get it from `ClassNotFoundException`.
- Add new property to `BrokenJobDetail` - it should store name of the missing class. Name - e.g. `originClassName`.
- Add new method to `QuartzService` - e.g. `getDisplayedClassName` - it should return `originClassName` if provided `JobDetails` is `BrokenJobDetails` or standard `.getJobClass().getName()` otherwise.

Reply:
No. I meant `InvalidJobDetails` stores `originClassName` - as a String, not Class.
And suggested `QuartzService#getDisplayedClassName(JobDetails)` extracts class name as a String - `originClassName` in case it's `InvalidJobDetail`, or `.getJobClass().getName()` otherwise.

This approach allow to firstly build JobDetails completely and then process in into JobModel.
Current approach make us create JobDetails and JobModel in parallel.